content/guides/development.md

```diff
@@ -5,182 +5,247 @@ contributors:
+print.bundle.js    6.43 kB       1  [emitted]         print
+     index.html  248 bytes          [emitted]
+   [0] ./src/print.js 84 bytes {0} {1} [built]
+   [1] ./src/index.js 403 bytes {0} [built]
+   [3] (webpack)/buildin/global.js 509 bytes {0} [built]
+   [4] (webpack)/buildin/module.js 517 bytes {0} [built]
+    + 1 hidden module
+Child html-webpack-plugin for "index.html":
+       [2] (webpack)/buildin/global.js 509 bytes {0} [built]
+       [3] (webpack)/buildin/module.js 517 bytes {0} [built]
+        + 2 hidden modules
 ```
 
-You may find it more convenient to run `serve` using npm scripts. You can do so by first creating a `start` script in `package.json` as follows:
+ Now open the resulting `index.html` file in your browser. Click the button and look in your console where the error is displayed. The error should say something like this:
 
-```json
-...
-"scripts": {
-  "start": "serve"
-}
-...
-```
+ ```
+ Uncaught ReferenceError: cosnole is not defined
+    at HTMLButtonElement.printMe (print.js:2)
+ ```
 
-You can then start the server by running `npm start` from within your project directory. After each compilation, you will need to manually refresh your browser to see the changes.
+ We can see that the error also contains a reference to the file (`print.js`) and line number (2) where the error occurred. This is great, because now we know exactly where to look to fix the issue.
 
-T> You may find the `--single` option useful for serving single page apps.
 
+## Choosing a Development Tool
 
-### Watch Mode with Chrome DevTools Workspaces
+W> Some text editors have a "safe write" function that might interfere with some of the following tools. Read [`Adjusting Your text Editor`](#adjusting-your-text-editor) for a solution to these issue.
 
-If you set up Chrome to [persist changes when saving from the _Sources_ panel](https://medium.com/@rafaelideleon/webpack-your-chrome-devtools-workspaces-cb9cca8d50da)
-so you don't have to refresh the page, you will have to setup webpack to use
+It quickly becomes a hassle to manually run `npm run build` everytime you want to compile your code.
 
-```javascript
-devtool: "inline-source-map"
-```
+There are a couple of different options available in webpack that help you automatically compile your code whenever it changes:
 
-to continue editing and saving your changes from Chrome or source files.
+ 1. webpack's Watch Mode
+ 2. webpack-dev-server
+ 3. webpack-dev-middlware
 
-There are some _gotchas_ about using workspaces with watch:
+In most cases, you probably would want to use `webpack-dev-server`, but let's explore all of the above options.
 
-* Large chunks (such as a common chunk that is over 1MB) that are rebuilt could cause the page to blank, which will force you to refresh the browser.
-* Smaller chunks will be faster to build than larger chunks since `inline-source-map` is slower due to having to base64 encode the original source code.
 
+### Using Webpack Watch Mode
```

Comment:
I would leave "Webpack" out here, above, and below... probably `webpack --watch` in the list, "Using Watch Mode" in the title and on line 105 just saying:

> You can instruct webpack to "watch" all files within your dependency graph for changes. If one of these files is updated, the code will be recompiled so you don't have to run the full build manually.